### PR TITLE
Add ability to rotate logs by size or date

### PR DESCRIPTION
**Pull request by HughePaul**

### CODE REVIEW

File: test/spec/spec.filerotatetransport.js [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+        afterEach(function () {
+            async.each.restore();
+            FileRotateTransport.prototype._dateRotateGlob.restore();
+        });
+
+        it('calls the callback without listing files if maxFiles is zero', function () {
+            transport.maxFiles = 0;
+            transport._dateRotateRemoveOldFiles(cb);
+            cb.should.have.been.calledWithExactly();
+            FileRotateTransport.prototype._dateRotateGlob.should.not.have.been.called;
```
Comment (gavboulton) [inline test/spec/spec.filerotatetransport.js]: The test doesn't describe this assertion. Does it need to be here?

File: test/spec/spec.filerotatetransport.js [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+        });
+
+        it('calls glob with a patern based on the logname', function () {
```
Comment (gavboulton) [inline test/spec/spec.filerotatetransport.js]: patern should be pattern.

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,5 @@
 language: node_js
 node_js:
-  - "0.10"
```
Comment (gavboulton) [inline .travis.yml]: Is there another reason for dropping 0.10, other than it's old? As this'll mean a major version bump should we also drop 0.12?

File: lib/filerotatetransport.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    var self  = this;
+    if (!this.opening && this.dateRotate && Date.now() >= this._dateRotateEndTime) {
+        debug('Needs new log file');
+
+        // Don't allow another open() while rotating files.
+        // This also will get set in _createStream
+        this.opening = true;
+
+        this._dateRotateLog(function () {
+            self._dateRotateRemoveOldFiles(function () {
```
Comment (HughePaul) [inline lib/filerotatetransport.js]: This was in the style of the parent file transport, but yes  i can update to be in a binding style instead

File: test/spec/spec.filerotatetransport.js (hunk 1 above)
Comment (HughePaul) [inline test/spec/spec.filerotatetransport.js]: when maxFiles is zero old files are not deleted, all rotated logs are left on disk, i.e. glob async isn't called to list files to delete.

File: test/spec/spec.filerotatetransport.js [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+
+        beforeEach(function () {
+            clock = sinon.useFakeTimers(Date.parse(new Date(2016, 10, 25, 12, 11, 10)));
+        });
+
+        afterEach(function () {
+            clock.restore();
+        });
+
+        it('should call the super constructor', function () {
```
Comment (HughePaul) [inline test/spec/spec.filerotatetransport.js]: 'should call the parent class constructor'?

Comment (gavboulton) [inline test/spec/spec.filerotatetransport.js]: Or 'should call the parent method'?

File: test/spec/spec.filerotatetransport.js (hunk 1 above)
Comment (gavboulton) [inline test/spec/spec.filerotatetransport.js]: This sounds like a good candidate for another test.

File: .travis.yml (hunk 3 above)
Comment (gavboulton) [inline .travis.yml]: Let's drop all 0.x versions.

Comment (HughePaul) [inline .travis.yml]: I've removed old node versions from travis. It now only tests against 4.x and 6.x

File: test/spec/spec.filerotatetransport.js (hunk 5 above)
Comment (HughePaul) [inline test/spec/spec.filerotatetransport.js]: Changed to parent method

File: test/spec/spec.filerotatetransport.js (hunk 1 above)
Comment (HughePaul) [inline test/spec/spec.filerotatetransport.js]: I've reworded the test description to better represent what it is testing

**Pull request merged**